### PR TITLE
Expand TA feature information in docs, and add subsections to left nav

### PR DESCRIPTION
**Pull request by meiqimichelle**
This PR creates separate pages for Test Analytics feature information so we can provide more detail into what we now know are commonly-asked questions. It also lays the groundwork for a future PR to pull env var info into its own section by starting to add subsections to the left hand nav. //cc @ddbeck

Communication goals, first draft in this PR:
- [x] Provide more info about our monitor options
- [x] Explain reliability statistic calculation
- [x] Explain why a build might be red in Test Analytics, and green in Pipelines (passes on retry will still be red in TA)
- [x] Explain what our p values mean
- [x] Confirm that TA works with parallel builds, and clarify the needed env vars to make this work correctly
- [x] Explain what our default branch is, and how to sneaky-access more (until we build into UI)
- [x] Explain that a test suite doesn't necessarily need to == all tests on a pipeline
- [ ] ~~Stretch goal: explain how to use our annotation feature (may need to come in a separate PR because MHz doesn't know how to do this yet)~~

~~NB: This is branched off of the new landing page work in https://github.com/buildkite/docs/pull/1523 because my brain couldn't handle thinking about this re-org without the new landing page in place.~~ After #1523 merged, I rebased to main. Should be good to go now. //cc @olyism @karensawrey 

Sending over to the docs team for review -- I imagine you'll have suggestions on how best to organize, and clearer sentences.

### CODE REVIEW

File: app/views/pages/index.html.erb [hunk 1]
```diff
@@ -37,9 +37,9 @@
         <span>Find and fix flaky tests</span>
         <ul>
           <li><%= link_to "Overview", "/docs/test-analytics" %></li>
-          <li><%= link_to "Deep performance analysis", "/docs/test-analytics#spot-sneaky-slow-downs" %></li>
-          <li><%= link_to "Find and fix flaky tests", "/docs/test-analytics#find-flaky-tests" %></li>
-          <li><%= link_to "Speed and reliability monitoring", "/docs/test-analytics#monitor-what-matters" %></li>
+          <li><%= link_to "Deep performance analysis", "/docs/test-analytics/features#spot-sneaky-slow-downs" %></li>
+          <li><%= link_to "Find and fix flaky tests", "/docs/test-analytics/features#find-flaky-tests" %></li>
```
Comment (olyism) [inline app/views/pages/index.html.erb:41]:
```suggestion
          <li><%= link_to "Find and fix flaky tests", "/docs/test-analytics/features#find-flaky-unreliable-tests" %></li>
```

🚨 Fix test for broken section/anchor link, as the heading has been changed on `/test-analytics/features`

**Review by olyism — CHANGES_REQUESTED** [review]
Thanks @meiqimichelle for putting the extra pages together. 

The build has failed on a missing section/anchor link on the homepage as the ["flaky test" heading](https://github.com/buildkite/docs/pull/1532/files#diff-61970285059c3fc9b5fe2d5b51fe729898101c7a0b3bbefff60f909badf10ed5R41) has changed. [Please view suggestion](https://github.com/buildkite/docs/pull/1532/files#diff-61970285059c3fc9b5fe2d5b51fe729898101c7a0b3bbefff60f909badf10ed5R41).

**Comment by olyism** [comment]
Thanks @meiqimichelle for putting the content together 🙌

I've got some thoughts on the design layout/visual styling, but as the PR is getting quite big so I'll raise a separate PR for design improvements.

**Comment by plaindocs** [comment]
Thanks for putting this together @meiqimichelle, there is a bunch of great stuff. Take a look at b330ece and see if it all makes sense to you? 

Environment variables refactor is now happening in #1536, it'll add a CI Environments page (which also needs adding to `beta_pages.rb` @ddbeck 

I don't love the feature page, as a mix of description of what it does, but I'm not sure of an alternative right now. 
I also don't love a ton of "Coming soon" sections, but as long as we make sure we do deliver those, and definitely don't give time estimates, it's hopefully useful info to some folks.

File: pages/test_analytics/features.md.erb [hunk 2]
```diff
@@ -1,33 +1,63 @@
...
+
+<%= image "test-trend.png", width: 1167, height: 394, alt: "Screenshot of test trend page showing change in duration across test runs and a recent failed test executions" %>
 
-## Monitor what matters
+Select any individual test execution to see more trend and deep-dive information, including comparisons against previous executions of this test.
 
-Configure monitors to track only what matters to you. Track reliability (percentage of passed test executions) over a specific number of days, or a specific number of test suite runs, and get alerts when a specific branch or or all branches fall below your defined threshold. Track duration changes against your test suite median, or number of consecutive failures.
+For example, in the following example, you can see that the test execution duration of 5.26 seconds is 2 seconds and 233 milliseconds more than the median of the previous 25 executions. In addition, the current value for this test execution of 5.26 seconds is in the 66th percentile within the distribution over the last 25 executions.
```
Comment (plaindocs) [inline pages/test_analytics/features.md.erb:46]:
```suggestion
For example, in the following screenshot, you can see that the test execution duration of 5.26 seconds is 2 seconds and 233 milliseconds more than the median of the previous 25 executions. In addition, the current value for this test execution of 5.26 seconds is in the 66th percentile within the distribution over the last 25 executions.
```

**Comment by meiqimichelle** [comment]
@plaindocs love your edits, thank you!

Agree on the "it's a bit odd to shove all this info into a page called 'Features'" -- it is a bit of a grab bag that has leftover vibes from when these docs were quite marketing-heavy. Maybe a refactor will become obvious once these changes are out the door.

I also don't love the 'coming soon' sections. The main one I'd like to tell people about (and I think why I started even writing 'coming soon' in the first place) was the bit about being able to URL hack other branch views into Test Analytics (even though you can only select 'all branches' or your default branch in the UI). The others aren't really as vital in terms of how to use the thing. 🤔 Should we just remove all of those sections for now?

**Comment by meiqimichelle** [comment]
Ugh. I'm re-reading, and comparing to our approach in other sections. It just doesn't match. This may be the best we can do for a first go -- but my goodness. I'm going to have a quick go based on your changes to move us further away from 'marketing' and more towards 'this is how you use the thing' -- I think that will help better align with the rest of our docs.

**Comment by meiqimichelle** [comment]
@plaindocs have a look at my refactor -- I didn't change the `_sidebar` file yet, but in this scenario I'd suggest re-naming that to 'Configure Test Suites' or somesuch.

**Comment by plaindocs** [comment]
@meiqimichelle love your work! I've edits the edits a nudge, done the sidebar work. Expecting these tests to pass. ;-)

**Comment by plaindocs** [comment]
I propose merging this, and starting any new edits in a new PR. 

(Incidentally we now have to edit a **bunch** of places when we change file or section names, I should probably write this up as a post, cos it's not gonna scale I don't think)

**Review by plaindocs — APPROVED** [review]
Good enough?

**Review by olyism — APPROVED** [review]

**Comment by olyism** [comment]
LGTM, any design tweaks can come in another PR so won't hold up the docs content

**Comment by meiqimichelle** [comment]
Awesome! Thanks for the further edits @plaindocs -- merging so that we're ready with any further improvements in another PR! 🚀

**Pull request merged**